### ISSUE TITLE
failed to query wlan interface, but iw and ip and ifconfig all show it

### ISSUE THREAD

**Issue by raingloom** [jaagr/polybar#1276]
I'm using the default setup from Arch that I copied to a PostmarketOS phone, the wifi interface is `wlan0` and it is working fine, but polybar fails to recognize it.

I have modified the config accordingly and replaced `wlp3s0` with `wlan0`, so that is not the problem.

**Comment by patrick96** [comment]
You want to run polybar on a phone? Interesting :thinking:

Can you post the whole output that polybar displays in the terminal (preferably when you ran it with `-l trace` or `-l info`).
Please also post the output of `ip link` and `iwconfig`

**Comment by raingloom** [comment]
```
$ ip link
1: lo: <LOOPBACK> mtu 16436 qdisc noop state DOWN 
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: ifb0: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN qlen 32
    link/ether 4a:c8:a8:ac:11:22 brd ff:ff:ff:ff:ff:ff
3: ifb1: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN qlen 32
    link/ether ba:bf:c7:5b:43:64 brd ff:ff:ff:ff:ff:ff
4: sit0: <NOARP> mtu 1480 qdisc noop state DOWN 
    link/sit 0.0.0.0 brd 0.0.0.0
5: ip6tnl0: <NOARP> mtu 1452 qdisc noop state DOWN 
    link/tunnel6 00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00 brd 00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00
6: rmnet0: <POINTOPOINT,MULTICAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ppp 
7: rmnet1: <POINTOPOINT,MULTICAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ppp 
8: rmnet2: <POINTOPOINT,MULTICAST,NOARP> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ppp 
9: p2p0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN qlen 1000
    link/ether a2:0b:ba:e7:f1:73 brd ff:ff:ff:ff:ff:ff
10: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN qlen 1000
    link/ether a0:0b:ba:e7:f1:73 brd ff:ff:ff:ff:ff:ff
11: rndis0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP qlen 1000
    link/ether 1e:0c:38:2b:d9:1c brd ff:ff:ff:ff:ff:ff
```

there is no `iwconfig` on the default install and idk which package it is in in Alpine, i'll check later

`-l trace` says that trace logging is not enabled

`-l info` output:

```
[1;32m* [0mLoaded /home/user/.config/polybar/config[0m

[1;33mwarn: [0mNo monitor specified, using "default"[0m
[1;32m* [0mLoaded monitor default (720x1280+0+0)[0m

[1;32m* [0mBar geometry: 720x35+0+0[0m

[1;31merror: [0mDisabling module "i3" (reason: The reference ${module/bspwm.label-focused-padding} does not exist (no fallback set))[0m

[1;31merror: [0mmodule/xbacklight: Could not get data (err: XCB_NAME (15))[0m
[1;31merror: [0mDisabling module "xbacklight" (reason: Not supported for "default")[0m

[1;31merror: [0mDisabling module "volume" (reason: Cannot find simple element)[0m

[1;31merror: [0mDisabling module "battery" (reason: No suitable way to get current charge state)[0m
[1;31merror: [0mDisabling module "temperature" (reason: The file '/sys/class/thermal/thermal_zone0/temp' does not exist)[0m

[1;32m* [0mStarting application[0m
[1;32m* [0mStarting module/filesystem[0m

[1;32m* [0mStarting module/memory[0m

[1;32m* [0mStarting module/cpu[0m

[1;32m* [0mStarting module/wlan[0m

[1;32m* [0mStarting module/date[0m

[1;32m* [0mStarting module/powermenu[0m

[1;32m* [0mEntering event loop (thread-id=1)[0m

[1;32m* [0mEventqueue worker (thread-id=4)[0m
[1;33mwarn: [0mmodule/wlan: Failed to query interface 'wlan0'[0m

[1;33mwarn: [0mmodule/wlan: Failed to query interface 'wlan0'[0m

[1;32m* [0mConfigured DPI = 96x96[0m

[1;32m* [0mLoaded font "fixed:pixelsize=10" (name=DejaVu Sans, offset=1, file=/usr/share/fonts/ttf-dejavu/DejaVuSans.ttf)[0m

[1;32m* [0mLoaded font "unifont:fontformat=truetype:size=8:antialias=false" (name=DejaVu Sans, offset=0, file=/usr/share/fonts/ttf-dejavu/DejaVuSans.ttf)[0m
[1;32m* [0mLoaded font "siji:pixelsize=10" (name=DejaVu Sans, offset=1, file=/usr/share/fonts/ttf-dejavu/DejaVuSans.ttf)[0m

[1;32m* [0mBar window: 0x0800002[0m

[1;33mwarn: [0mIgnoring restack of i3 window (not needed when `override-redirect = false`)[0m

[1;32m* [0mActivating tray manager[0m

[1;32m* [0mTray window: 0x0800009[0m

[1;32m* [0mmodule/filesystem: Rebuilding cache[0m

[1;32m* [0mmodule/memory: Rebuilding cache[0m
[1;32m* [0mmodule/cpu: Rebuilding cache[0m

[1;32m* [0mmodule/wlan: Rebuilding cache[0m

[1;32m* [0mmodule/date: Rebuilding cache[0m

[1;32m* [0mmodule/powermenu: Rebuilding cache[0m

[1;32m* [0mRedrawing bar window[0m

[1;33mwarn: [0mDropping unmatched character  (U+e028)[0m
[1;33mwarn: [0mDropping unmatched character  (U+e026)[0m

[1;33mwarn: [0mDropping unmatched character  (U+e016)[0m

[1;33mwarn: [0mDropping unmatched character  (U+e10c)[0m

[1;32m* [0mNotifying pending tray clients[0m

[1;32m* [0mmodule/memory: Rebuilding cache[0m

[1;32m* [0mmodule/cpu: Rebuilding cache[0m
[1;32m* [0mRedrawing bar window[0m

[1;33mwarn: [0mDropping unmatched character  (U+e028)[0m
[1;33mwarn: [0mDropping unmatched character  (U+e026)[0m

[1;33mwarn: [0mDropping unmatched character  (U+e016)[0m
[1;33mwarn: [0mDropping unmatched character  (U+e10c)[0m

[1;33mwarn: [0mmodule/wlan: Failed to query interface 'wlan0'[0m

[1;32m* [0mmodule/memory: Rebuilding cache[0m

[1;32m* [0mmodule/cpu: Rebuilding cache[0m
[1;32m* [0mRedrawing bar window[0m

[1;33mwarn: [0mDropping unmatched character  (U+e028)[0m
[1;33mwarn: [0mDropping unmatched character  (U+e026)[0m

[1;33mwarn: [0mDropping unmatched character  (U+e016)[0m

[1;33mwarn: [0mDropping unmatched character  (U+e10c)[0m

[1;33mwarn: [0mmodule/wlan: Failed to query interface 'wlan0'[0m

[1;32m* [0mmodule/memory: Rebuilding cache[0m

[1;32m* [0mmodule/cpu: Rebuilding cache[0m
[1;32m* [0mRedrawing bar window[0m

[1;33mwarn: [0mDropping unmatched character  (U+e028)[0m

[1;33mwarn: [0mDropping unmatched character  (U+e026)[0m
[1;33mwarn: [0mDropping unmatched character  (U+e016)[0m

[1;33mwarn: [0mDropping unmatched character  (U+e10c)[0m

[1;32m* [0mmodule/memory: Rebuilding cache[0m

[1;32m* [0mmodule/cpu: Rebuilding cache[0m

[1;33mwarn: [0mmodule/wlan: Failed to query interface 'wlan0'[0m

[1;32m* [0mmodule/memory: Rebuilding cache[0m
[1;32m* [0mmodule/cpu: Rebuilding cache[0m
```

(yeah, i'm using a weird terminal)

**Comment by patrick96** [comment]
According to the [Wiki](https://wiki.alpinelinux.org/wiki/Connecting_to_a_wireless_access_point), `iwconfig` should be part of `wireless_tools` or `wpa_supplicant`.

**Comment by NBonaparte** [comment]
Closing due to inactivity. This should also be fixed by #1009.
